htlcswitch/spider_test.go

```diff
@@ -9,11 +9,10 @@ import (
 )
 
 // bob<->alice channel has insufficient BTC capacity/bandwidth. In this test we
-// sduration the payment from Carol to Alice over Bob peer. (Carol -> Bob -> Alice)
-// Right now this payment returns an error immediately. Instead, we want it to
-// be queued on Bob -> Alice channel, until Alice sdurations a payment upstream to
-// Bob, and then this payment should be processed.
-func TestSpiderInsufficentFunds (t *testing.T) {
+// send the payment from Alice to Carol over the Bob peer. (Alice -> Bob ->
+// Carol) with the Bob -> Carol link being the bottleneck.
+// After Carol sends Bob money too, this payment should go through.
+func TestSpiderTemporarilyInsufficentFunds(t *testing.T) {
```

Comment:
is the flow here basically that you schedule this thread to send funds fromCarol -> Bob x seconds later and first try to send Alice -> Carol which will wait on funds from Bob -> Carol. Once the initial thread goes through x seconds later, the Alice-> Carol transaction succeeds and all is well? If so, can you maybe reorder the functions such that that is apparent or comment on the goroutine itself?

Reply:
fixed (changed the order these functions are called so Alice -> Carol gets called first).